core/services/streams/stream_registry.go

```diff
@@ -39,29 +44,52 @@ func newRegistry(lggr logger.Logger, runner Runner) *streamRegistry {
 		sync.RWMutex{},
 		lggr.Named("Registry"),
 		runner,
-		make(map[StreamID]Stream),
+		make(map[StreamID]Pipeline),
+		make(map[int32]Pipeline),
 	}
 }
 
-func (s *streamRegistry) Get(streamID StreamID) (strm Stream, exists bool) {
+func (s *streamRegistry) Get(streamID StreamID) (p Pipeline, exists bool) {
```

Comment:
What's the relation between job ids, stream ids, and pipeline ids? Seems like job and stream are analogous, and then 1 pipeline can have many streams (which is a specific type of job).  Is that correct?

Reply:
Yeah this is a bit confusing. A `job` has exactly one `pipeline_spec` and those have two different IDs (we almost exclusively use the `jobID` to reference it though). `stream_id` is a completely different thing. So a job has one pipeline which may contain many stream IDs.